src/authentication-strategies/JWT.strategy.ts

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+import {promisify} from 'util';
+const verifyAsync = promisify(jwt.verify);
+// Consider turn it to a binding
+const SECRET = 'secretforjwt';
+import {Request, HttpErrors} from '@loopback/rest';
+import {UserProfile} from '@loopback/authentication';
+import * as _ from 'lodash';
+import {AuthenticationStrategy} from './authentication.strategy';
+
+export class JWTStrategy implements AuthenticationStrategy {
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    // there is a discussion regarding how to retrieve the token,
+    // see comment https://github.com/strongloop/loopback-next/issues/1997#issuecomment-451054806
+    const token = request.query.token || request.headers['authorization'];
+    if (token) {
```

Comment:
Please use reverse logic and early return to keep the code easier to read and less indented.

```ts
if (!token) {
  throw new HttpErrors.Unauthorized('Token not found!');
}
```